internal/kubelet/client_test.go

```diff
@@ -102,15 +103,28 @@ func TestDefaultTLSClient(t *testing.T) {
 }
 
 func TestSvcAcctClient(t *testing.T) {
+	server := httptest.NewUnstartedServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		// Check if call is authenticated using token from test file
+		require.Equal(t, req.Header.Get("Authorization"), "Bearer s3cr3t")
+		_, err := rw.Write([]byte(`OK`))
+		require.NoError(t, err)
+	}))
+	cert, err := tls.LoadX509KeyPair("./testdata/testcert.crt", "./testdata/testkey.key")
+	require.NoError(t, err)
+	server.TLS = &tls.Config{Certificates: []tls.Certificate{cert}}
+	server.StartTLS()
+	defer server.Close()
+
 	p := &saClientProvider{
-		endpoint:   "localhost:9876",
-		caCertPath: certPath,
+		endpoint:   server.Listener.Addr().String(),
+		caCertPath: "./testdata/testcert.crt",
 		tokenPath:  "./testdata/token",
 		logger:     zap.NewNop(),
 	}
-	cl, err := p.BuildClient()
+	client, err := p.BuildClient()
+	resp, err := client.Get("/")
 	require.NoError(t, err)
-	require.Equal(t, "s3cr3t", string(cl.(*clientImpl).tok))
```

Comment:
`NewBearerAuthWithRefreshRoundTripper` handles the authorization header for us, so removing this test case and replacing with the server validation check